hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,249 @@
+                    .skip(1)
+                    .map(line -> parseCSVLine(line, columnCount))
+                    .toList();
+        } catch (IllegalArgumentException iae) {
+            log.error("Failed to parse blocklist, entry does not have required number of columns", iae);
+            return;
+        } catch (DecoderException de) {
+            log.error("Failed to parse blocklist, entry not in hex format", de);
+            return;
+        } catch (Exception e) {
+            log.error("Failed to read blocklist file {}", blocklistFileName, e);
+            return;
+        }
+
+        final var blockedToCreate = blocklist.stream()
+                .filter(blockedAccount ->
+                        aliasManager.lookupIdBy(blockedAccount.evmAddress).equals(MISSING_NUM))
+                .collect(Collectors.toSet());
+
+        final var genesisAccountId = AccountID.newBuilder()
+                .setRealmNum(0)
+                .setShardNum(0)
+                .setAccountNum(accountNumbers.treasury())
+                .build();
+
+        for (final var blockedInfo : blockedToCreate) {
+            final var newId = ids.newAccountId(genesisAccountId); // get the next available new account ID
+            final var account = blockedAccountWith(blockedInfo);
+            accounts.put(newId, account); // add the account with the corresponding newId to state
+            accountsCreated.add(account); // add the account to the list of accounts created by this class
+            aliasManager.link(
+                    blockedInfo.evmAddress,
+                    EntityNum.fromAccountId(newId)); // link the EVM address alias to the new account ID
+        }
+    }
+
+    private List<String> readPrivateKeyBlocklist(String fileName) {
+        final var inputStream = getClass().getClassLoader().getResourceAsStream(fileName);
+        final var reader = new BufferedReader(new InputStreamReader(inputStream));
+        return reader.lines().toList();
+    }
+
+    /**
+     * Parses a line from the blocklist file and returns blocked account info record.
+     *
+     * The line should have the following format:
+     * <private key>,<memo>
+     *     where <private key> is a hex-encoded private key
+     *     and <memo> is a memo for the blocked account
+     *     and both values are comma-separated.
+     *
+     * The resulting blocked account info record contains the EVM address derived from the private key, and the memo.
+     *
+     * @param line line from the blocklist file
+     * @param columnCount number of comma-separated values in a line
+     * @return blocked account info record
+     */
+    private BlockedInfo parseCSVLine(String line, int columnCount) {
+        final var parts = line.split(",", -1);
+        if (parts.length != columnCount) {
+            throw new IllegalArgumentException("Invalid line in blocklist file: " + line);
+        }
+
+        final var privateKeyBytes = Hex.decode(parts[0]);
```

Comment:
(this is where you could trap the exception for bad hex and throw instead an exception that contained the actual bad line so the user could know)